components/UserObserver.php

```diff
@@ -122,7 +126,7 @@ private function getUserAuthentication(string $username): UserAuthentication
 	{
 		$criteria = new \CDbCriteria();
 		$criteria->addCondition('username = :username');
-		$criteria->params[':username'] = "WILLIAMSS";//$params['username'];
+		$criteria->params[':username'] = $username;
 		$user_authentication = \UserAuthentication::model()->find($criteria);
```

Comment:
Sorry for forgetting to mention this in the previous review, but there should be similar criteria for UserAuthentication as the one for User. The `\UserAuthentication::model()->find($criteria);` should not cause exceptions but it will only fetch the first record, which potentially will not be the desired one.

Reply:
I've added the institution_authentication_id